### PR TITLE
[Lead space] website components - draft update leadspace.mdx

### PR DESCRIPTION
**Pull request by DragosRistici**
### Related Ticket(s)

{{Provide url(s) to the related ticket(s) that this pull request addresses}}

### Description

{{Add a human-readable description / detail summary of what the PR is changing and any details around how and why}}

{{If applicable, include a screenshot indicating an example or examples of what the PR is changing in the application}}

### Changelog

**New**

- {{new thing}}

**Changed**

- {{changed thing}}

**Removed**

- {{removed thing}}

### CODE REVIEW

**Comment by shixiedesign** [comment]
A few general comment: 
1. I'm confused about when content goes to "Pattern" page and when it goes to "Components" page. Usage info such as height variants should go here, no? 
2. React icon on the card broke 
image

File: src/pages/components/leadspace.mdx [hunk 1]
```diff
@@ -29,7 +28,23 @@ import ComponentFeedback from "components/ComponentFeedback";
 
 This is the most commonly used Lead space style. The left-aligned type draws the users' attention to the Lead space messaging immediately. Used in marketing and product pages, among others.
 
-![Lead space left-aligned](../../images/pattern/leadspace/Leadspace_Demo_Centered.jpg)
+There are four available height sizes:
```
Comment (shixiedesign) [inline src/pages/components/leadspace.mdx:31]: There are 4 available heights:

Comment (shixiedesign) [reply src/pages/components/leadspace.mdx:31]: I think.... ^ did anyone check IBM Styles guidance on this?

File: src/pages/components/leadspace.mdx [hunk 2]
```diff
@@ -29,7 +28,23 @@ import ComponentFeedback from "components/ComponentFeedback";
...
+
+- Super (640px)
+- Tall (560px)
+- Medium (480px)
+- Short (320px)
+
+For detailed guidelines around usage, also reference the [Lead space pattern page](/patterns/lead-space).
```
Comment (shixiedesign) [inline src/pages/components/leadspace.mdx:38]: "For specific usage of each height, please [see Lead space pattern](/patterns/lead-space) page. "

Generally we should make sure <a> link wrap the the Call to action - always try to start with a verb / action. Good for accessibility. Imagine screen reader voice.

File: src/pages/components/leadspace.mdx [hunk 3]
```diff
@@ -41,24 +56,27 @@ This option has 4 empty columns on the left and right to accommodate pages that
...
-| Primary call to action   | No                                                                                                    | XXS                                       |
-| Background image         | Yes<br/> <br/><div style="max-width: 200px"> Note: each breakpoint requires a different image.</div> |                                           |
+| Field          | Field type      | Required                              | Max character size <br/>(English / translated) |  Notes          |
+| -------------- | --------------- | ------------------------------------- | ---------------------------------------------- | --------------- |
+| Headline       | Short copy      | Yes                                   | XS (65 / 85)                                   | –               |
+| Description    | Short copy      | No                                    | M (120 / 150)                                  | –               |
+| [Button group](https://www.ibm.com/standards/web/carbon-for-ibm-dotcom/components/button-group) | Component | No | XXS (40 / 55) | –        |
```
Comment (shixiedesign) [inline src/pages/components/leadspace.mdx:63]: Maybe mention button groups will not be available on `Short`? I think we decided this. Lemme know if we didn't and I was dreaming.

File: src/pages/components/leadspace.mdx [hunk 4]
```diff
@@ -70,23 +88,28 @@ For detailed guidelines around usage, also reference the [Lead space pattern pag
...
-![Light theme](../../images/pattern/leadspace/LeadSpace_ColorThemes_Light.jpg)
+### Light theme
+
+Use `Light (white)` when the background image is well-lit or brightly colored. The gradient overlay will render white and the text will be gray-100.
 
-<Caption>Light</Caption>
+![Light_theme](../../images/pattern/leadspace/leadspace_colortheme_light_demo.jpg)
 
-## Left-aligned style options
+<Caption>Light Lead space – Super – 640px height at xlg 1312px break point</Caption>
```
Comment (shixiedesign) [inline src/pages/components/leadspace.mdx:106]: breakpoint <-- no space

File: src/pages/components/leadspace.mdx [hunk 5]
```diff
@@ -70,23 +88,28 @@ For detailed guidelines around usage, also reference the [Lead space pattern pag
...
 
-### Type styles
+## Type styles
 
-There are two expressive type styles to choose from when using Lead space left-aligned:
+### Lead space left-aligned
+
+There are two recommended expressive type styles to choose from when using Lead space left-aligned:
```
Comment (shixiedesign) [inline src/pages/components/leadspace.mdx:112]: Thanks for adding this though I was hoping we can...one day...narrow down to 1 type style per height. But for now, totally accurate.

File: src/pages/components/leadspace.mdx [hunk 6]
```diff
@@ -107,70 +130,109 @@ There are two expressive type styles to choose from when using Lead space left-a
...
-1. **Button group:** Use the Button group component
-1. **Gradient stop:** color: black (#000000), opacity: 75%, linear: left to right, width: 50% of total container width
-1. **Background image:** \*optional, width: 100% of container for all breakpoints. For detailed guidelines, visit [the imagery section on the Lead space pattern page](/patterns/lead-space#imagery).
+1. **Heading** <br/>Type: $expressive-heading-06, color: $text-01, width: calc(8col – 32px)
+1. **Content spacer** <br/>Minimum space between lead space heading and short copy: 96px
+1. **Short copy** \*optional <br/>Type: $expressive-heading-03, color: $text-01, width: calc(6col – 32px)
+1. **Button group** <br/>Use the Button group component
+1. **Gradient (applied over image)** <br/>Color: black (#000000), linear: left to right <br/>Opacity: 1 at start, 0.4 at 50%, 0 at 75% of total container width
+1. **Background image** \*optional <br/>Cover, width: 100% of container for all breakpoints. For detailed guidelines, visit [the imagery section on the Lead space pattern page](/patterns/lead-space#imagery).
```
Comment (shixiedesign) [inline src/pages/components/leadspace.mdx:173]: I see why you aded the <br /> , esp. on Gradient line where there are 2 line items... I suspect there's a deeper IA problem though. If we wanna document specs, there should probably be a better way to do it. It's honestly also difficult to keep up to date. Nevertheless, this looks good.

**Review by shixiedesign — APPROVED** [review]
Thank you! 👍

File: src/pages/components/leadspace.mdx [hunk 7]
```diff
@@ -29,7 +28,22 @@ import ComponentFeedback from "components/ComponentFeedback";
 
 This is the most commonly used Lead space style. The left-aligned type draws the users' attention to the Lead space messaging immediately. Used in marketing and product pages, among others.
 
-![Lead space left-aligned](../../images/pattern/leadspace/Leadspace_Demo_Centered.jpg)
+There are 4 available heights:
+
+- Super (640px)
+- Tall (560px)
+- Medium (480px)
+- Short (320px)
```
Comment (photodow) [inline src/pages/components/leadspace.mdx:36]: Can we provide this information in a table? Is this only for left aligned?

File: src/pages/components/leadspace.mdx [hunk 8]
```diff
@@ -41,26 +55,27 @@ This option has 4 empty columns on the left and right to accommodate pages that
...
+| ------------------ | --------------- | ------------ | ----------------- | ------------------------------------------------------| ---------------|
+| Headline           | Short copy      | Yes          | 1                 | XS (65 / 85)                                          | –              |
+| Description        | Short copy      | No           | 1                 | M (120 / 150)                                         | –              |
+| [Button group](https://www.ibm.com/standards/web/carbon-for-ibm-dotcom/components/button-group) | Component | No | Max of two buttons | XXS (40 / 55) | –    |
+| Gradient           | True/False      | No           | –                 | –                                                     | True - Show gradient (default)<br/>False - No gradient |
+| Background [image](https://www.ibm.com/standards/web/carbon-for-ibm-dotcom/components/image)               | Component | No | 1                  | – | Breakpoints require different images.    |
+| Type               | Option list     | Yes          | –                 | –                                                     | Center, Left, or Small |
```
Comment (photodow) [inline src/pages/components/leadspace.mdx:76]: Are we overcomplicating things by providing two separate content models for a single component? I feel look at consolidating.

File: src/pages/components/leadspace.mdx [hunk 9]
```diff
@@ -41,26 +55,27 @@ This option has 4 empty columns on the left and right to accommodate pages that
 
 ### Lead space left-aligned
```
Comment (photodow) [inline src/pages/components/leadspace.mdx:56]: Lead space left maybe? We aren't saying the other one is leadspace center-aligned.

File: src/pages/components/leadspace.mdx [hunk 10]
```diff
@@ -70,23 +85,28 @@ For detailed guidelines around usage, also reference the [Lead space pattern pag
...
-<Caption>Dark</Caption>
+<Caption>Dark Lead space – Super – 640px height at xlg 1312px breakpoint</Caption>
 <br />
 
-![Light theme](../../images/pattern/leadspace/LeadSpace_ColorThemes_Light.jpg)
+### Light theme
+
+Use `Light (white)` when the background image is well-lit or brightly colored. The gradient overlay will render white and the text will be gray-100.
+
+![Light_theme](../../images/pattern/leadspace/leadspace_colortheme_light_demo.jpg)
```
Comment (photodow) [inline src/pages/components/leadspace.mdx:101]: This whole color theme section isn't clear to me. First Carbon doesn't have a light theme and a dark theme. Also, do we do this anywhere else within our components? All of our components are themeable aside like masthead and footer.

File: src/pages/components/leadspace.mdx [hunk 11]
```diff
@@ -107,70 +127,109 @@ There are two expressive type styles to choose from when using Lead space left-a
...
+<br />
+
+1. **Heading** <br/>Type: $display-01, color: $text-01, width: calc(8col – 32px)
+2. **Short copy** \*optional <br/>$expressive-heading-03, color: $text-01, width: calc(6col – 32px)
+3. **Button group** <br/>Use the Button group component
+4. **Content bottom spacer** <br/>Height: $layout-07 + $layout-06 with background image, <br/>or $layout-07 without background image
+5. **Gradient (applied over image)** <br/>Color: black (#000000), linear: top to bottom, <br/>Opacity: 1 at start, 0.6 at 256px from bottom, 0 at 160px from bottom
+6. **Background image** \*optional <br/>Cover, width: 100% of container for all breakpoints. <br/>Image position: center bottom.
 
-The design specs and functional specs for Lead space can be viewed <a href="https://ibm.box.com/s/d9jey3rlfxqmxu059f0x9dtcx7f5ehuh" target="_blank" rel="noreferrer">here</a>.
```
Comment (photodow) [inline src/pages/components/leadspace.mdx:173]: Are we adding functional and design specs to the website now? It looks like we're just duplicating our content model here again. I think we should link out to corresponding files like we do on our other pages.

File: src/pages/components/leadspace.mdx [hunk 12]
```diff
@@ -70,23 +85,28 @@ For detailed guidelines around usage, also reference the [Lead space pattern pag
...
+![Light_theme](../../images/pattern/leadspace/leadspace_colortheme_light_demo.jpg)
 
-<Caption>Light</Caption>
+<Caption>Light Lead space – Super – 640px height at xlg 1312px breakpoint</Caption>
 
-## Left-aligned style options
+## Type styles
 
-### Type styles
+### Lead space left-aligned
```
Comment (photodow) [inline src/pages/components/leadspace.mdx:108]: If we're going to have a hierarchy to our types dependent on specific values in other props (I would strongly suggest we try not to do this) then we need to represent that hierarchy in our documentation. left styles should be under left, and centered styles should be under centered above in my opinion. This information is kind of sitting out here on its own.

**Review by photodow — CHANGES_REQUESTED** [review]
- [ ] The wording here feels a little repetitive.

image

- [ ] Generally speaking, I think we should slow down on this, and only add what we have in production to this component documentation. What you have on this file is not reflective of what's in production, and in so doing that I would label it as coming soon because it's not here yet a lot of things still in motion.

image

- [ ] I think if we can fix some of the hierarchy on this page I think that would help a lot.

Thanks for working on this @DragosRistici!

File: src/pages/components/leadspace.mdx (hunk 7 above)
Comment (DragosRistici) [reply src/pages/components/leadspace.mdx:36]: for left-aligned only

Comment (DragosRistici) [reply src/pages/components/leadspace.mdx:36]: Added them in a small table, thank you, @photodow, for the idea!

File: src/pages/components/leadspace.mdx (hunk 8 above)
Comment (DragosRistici) [reply src/pages/components/leadspace.mdx:76]: We can definitely look at this. I already started it, but the stringent need for a shorter height and previous research already in place topped it.

File: src/pages/components/leadspace.mdx (hunk 9 above)
Comment (DragosRistici) [reply src/pages/components/leadspace.mdx:56]: I believe it is "left-aligned", "right-aligned", "justified" and "centered". The common part is the suffix "-ed".  But I don't believe "center-aligned" would be too wrong either, just misleading, as nothing in it is aligned to the center, so considering its position as centered, I do suggest, though, sacrificing the "center-aligned" possibility as too questionable. So you can say the lead space is centered, but still, its elements are left-aligned.

I think this discussion should be included in the next component overhaul.

File: src/pages/components/leadspace.mdx (hunk 10 above)
Comment (DragosRistici) [reply src/pages/components/leadspace.mdx:101]: That's true. It somehow was left hanging. I changed it now to reflect reality.

File: src/pages/components/leadspace.mdx (hunk 11 above)
Comment (DragosRistici) [reply src/pages/components/leadspace.mdx:173]: I agree with you, i adjusted the examples.

File: src/pages/components/leadspace.mdx (hunk 12 above)
Comment (DragosRistici) [reply src/pages/components/leadspace.mdx:108]: Let me know how do you want to proceed with this one.

**Comment by DragosRistici** [comment]
@photodow @JennySanchez 
1. How do we adjust this to reflect its WIP?
image
2. Jenny, please help me adjust this
image


Thank you!

File: src/pages/components/leadspace.mdx (hunk 12 above)
Comment (DragosRistici) [reply src/pages/components/leadspace.mdx]: Settled.

File: src/pages/components/leadspace.mdx (hunk 1 above)
Comment (JennySanchez) [reply src/pages/components/leadspace.mdx]: @shixiedesign 

`I'm confused about when content goes to "Pattern" page and when it goes to "Components" page. Usage info such as height variants should go here, no?`

Usage info such as height variants should go to the components page. The Patterns section contains high-level best practices when using components in combinations of other components, layouts, and flows that address common user objectives. Therefore, any tailored information at the component level will live on the components page.

"React icon on the card broke" 
Not sure how did this happen in preview, but it works fine on the [live website](https://www.ibm.com/standards/web/carbon-for-ibm-dotcom/components/leadspace#feedback).

Screen Shot 2021-04-13 at 1 50 50 PM

File: src/pages/components/leadspace.mdx (hunk 7 above)
Comment (JennySanchez) [reply src/pages/components/leadspace.mdx]: Agreed with @photodow  on **"Generally speaking, I think we should slow down on this, and only add what we have in production to this component documentation."**

Comment (JennySanchez) [reply src/pages/components/leadspace.mdx]: @photodow The Lead space page is one of the oldest components of websites, and we do need to work on the hierarchy, but this work will not be included in this PR.

File: src/pages/components/leadspace.mdx (hunk 9 above)
Comment (JennySanchez) [reply src/pages/components/leadspace.mdx:63]: I prefer "Lead space – Left" and "Lead space – Center". Note: Should we have the word "Lead space" in the naming?

**Pull request merged**